MetadataExtractor/Formats/QuickTime/QuickTimeReaderExtensions.cs

```diff
@@ -9,12 +9,11 @@ public static class QuickTimeReaderExtensions
     {
         public static string Get4ccString(this SequentialReader reader)
         {
-            var sb = new StringBuilder(4);
-            sb.Append((char)reader.GetByte());
-            sb.Append((char)reader.GetByte());
-            sb.Append((char)reader.GetByte());
-            sb.Append((char)reader.GetByte());
-            return sb.ToString();
+            Span<byte> bytes = stackalloc byte[4];
+
+            reader.GetBytes(bytes);
+
+            return Encoding.ASCII.GetString(bytes);
```

Comment:
This causes a behaviour change for some inputs, though I've only seen it on fuzzed files that contain very weird data:

![image](https://github.com/drewnoakes/metadata-extractor-dotnet/assets/350947/42766eb6-98e5-4986-ac84-459b02a4314e)

The ASCII encoding replaces some characters with `?` which potentially loses information. According to https://en.wikipedia.org/wiki/FourCC non-printable characters are valid. I'm not sure it's a problem in practice, but I think I'll make a change here to restore the old behaviour.

Reply:
#391